blog/content/post/2020-11-30-coreboot_for_power.md

```diff
@@ -0,0 +1,154 @@
+less time spent on reading from flash, but also more space for other components,
+like new drivers for Skiroot.
+
+#### For the coreboot community
+
+coreboot would get a **new architecture** supported. Although there are stubs,
+cross-compiler and libraries for PPC, the coreboot itself isn't in working state
+when it comes to this architecture as of now. It would also get a **new, truly
+open** ([unless you want SATA controller](https://wiki.raptorcs.com/wiki/PM8068))
+platform.
+
+PowerPC by default uses big endian. It is configurable at runtime, but the
+libraries included in coreboot-sdk are compiled for BE. This gives a perfect
+opportunity to validate that **coreboot works also for big endian platforms**.
+By now we confirmed that there are parts in CBFS and FMAP which need small
+fixes. The endianness of fields also needs to be properly documented for those
+structures, as they can be used both by a platform it is started on, as well as
+an application on a platform it is built on. Those two do not have to use the
+same endianness.
+
+The biggest achievement of this port would undoubtedly be bringing support for
+**native DDR4 initialisation and training**. Even though the details of
+accessing the hardware are platform-specific, the general process stays the
+same. One caveat might be that Talos II supports only RDIMM (registered, or
+buffered DIMM) and that is the only kind of memory sticks we are able to test,
+however even if it won't work out of the box for different DIMM flavours, it
+should give a good base to build on.
+
+## Current state of work
+
+We have successfully started coreboot on the platform:
+
+[![asciicast](https://asciinema.org/a/JQ1MaBSzGN1L1JcbgTX3G3kt6.svg)](https://asciinema.org/a/JQ1MaBSzGN1L1JcbgTX3G3kt6?speed=1)
+
+Actually, we were starting it much earlier than we knew it, because we used a
+wrong address for serial port. It was caused mostly because of improper analysis
+of Hostboot's code, so we ended up converting virtual address to a physical one
+using a negative offset compared to what we should have used. This is what you
+get from mixing virtual memory management and manual pointer calculations 🙂️
+
+Since then we believe we have [mostly fixed all of the endianness issues](https://github.com/3mdeb/coreboot/commits/fmap_cbfs_endian),
+so `Couldn't load romstage` became `Payload not loaded`, without any actual
+initialisation code. After thorough tests and updates to the documentation we
+will begin to upstream these changes.
+
+If you think we can help in improving the security of your firmware or you are
+looking for someone who can boost your product by leveraging advanced features
+of used hardware platform, feel free to [book a call with us](https://calendly.com/3mdeb/consulting-remote-meeting)
+or drop us email to `contact<at>3mdeb<dot>com`. If you are interested in similar
+content feel free to [sign up to our newsletter](http://eepurl.com/doF8GX)
```

Comment:
@krystian-hebel I would improve this summary in the context of the article e.g. we would be glad to support the enabling of new  RISC-V or POWER10 platforms.

I'm missing the community excitement that we experienced. Maybe @miczyg1 can add something. There are also notes from OSFC which can be shared and reflected in this post.

We have to keep the bar high.

Reply:
After the Power9, the Power10, then the power based laptop and finally an alternative to x86 to the actual exosystem by diversity of form factor choices, reduction of production costs. End user accessibility by cost factor. But first: openness for developers, better understanding of the architecture as a start.

I would try to find a way to point that yeah, actual prices of early adopters are high and require more radical people demanding a x86 alternative. But once the price of the boards go down, that CPUs become constructed not only for servers but for workstations and demand is there for laptop and lower power consumption, something else then the https://www.powerpc-notebook.org/en/ will be available. 

Something in the lines of: The Talos II is already RYF certified. It will now have coreboot, permitting payloads ecosystems to traverse x86 for performance and openness. Future ports to other boards will be eased. Costs of such boards will be reduced by demand.

Also.
By setting the bar higher for the coreboot supported platforms to finally have a real open source firmware existence (no PSP, no ME, no FSP, no e2dk) users and developers will finally have an option to choose from. Cloud providers will need to take a stance in terms of the platforms they choose in protecting their user's data, their privacy without choosing.

I do not know if we want to mix things and tackle the fact that QubesOS is taking small steps toward the server world (4.1 with headless remote support), that the network stack is taking necessary steps to be able to create DMZs (https://github.com/Rudd-O/qubes-network-server/), that Xen developers are ready to jump back into the PPC64 world, so that a real, strong alternative to x86 is showing the tip of its nose?

That the people actually choosing  virtualization platforms for their compatibility with kvm don't have to choose x86 anymore. That soon Xen will be part of the offer, and that now is the moment to show interest and support?

Not a marketing guy here as you see. But that is what I would love to read. Some hope getting of of x86 without having to recreate the wheel. Get that the ecosystem is transferred this time. So that other projects relying on coreboot can find their way into x86 alternatives without sacrificing performance/virtualization, nor choose between RAM availability. Privacy and blobless architecture is available now. The need for the common base for payload ecosystem will be done. And maybe even if bold, ask readers to say what they would love to see happen next!

Reply:
@tlaurion 100% agree. @krystian-hebel any chance you can interleave Thierry comments into a summary?